modes/preclinical-4d/src/index.tsx

```diff
@@ -91,14 +97,17 @@ function modeFactory({ modeConfiguration }) {
       // it may change the protocol/stage based on workflow stage settings
       initWorkflowSteps({ servicesManager });
     },
-    onModeExit: ({ servicesManager }: withAppTypes) => {
+    onModeExit: ({ servicesManager, extensionManager }: withAppTypes) => {
       const {
         toolGroupService,
         syncGroupService,
         segmentationService,
         cornerstoneViewportService,
       } = servicesManager.services;
 
+      // Turn 4D mode off to move back to its initial state.
+      volumeLoaderUtil.disable4DMode();
```

Comment:
This situation is somewhat concerning. What's the reason for it? If it's truly necessary, we should implement a service with an `onModeExit` method to ensure it always runs.

Reply:
There is a call to `volumeLoaderUtil.enable4DMode()` on line 55 in `onModeEnter` and this call to `disable4DMode()` here in `onModeExit` because only in this mode 4D will be enabled otherwise the viewer will try to split any series into a 4D series even in basic mode.